mod/ItemImpls/FillerAndTrap/SupernovaTrap.cs

```diff
@@ -0,0 +1,42 @@
+﻿using HarmonyLib;
+
+namespace ArchipelagoRandomizer
+{
+    [HarmonyPatch]
+    internal class SupernovaTrap
+    {
+        private static uint _supernovaTraps;
+
+        public static uint supernovaTraps
+        {
+            get => _supernovaTraps;
+            set
+            {
+                if (value > _supernovaTraps)
+                {
+                    TriggerSupernova();
+                }
+                _supernovaTraps = value;
+            }
+        }
+
+        private static bool triggeredSupernovaInThisLoop = false;
+
+        private static void TriggerSupernova()
+        {
+            //trigger ONLY in the main solar system scene. Supernova while reaching the eye is not a good idea
+            if (LoadManager.GetCurrentScene() != OWScene.SolarSystem)
+                return;
+
+            //prevents triggering twice if a supernova is already in progress- whether by this trap of by the timeloop
+            if (triggeredSupernovaInThisLoop || TimeLoop.GetSecondsRemaining() <= 0.0)
```

Comment:
note to self: double-check if this is the correct condition for _the supernova starting_, or if it's for the TimeLoop death that gets you even when you're out of supernova range

Reply:
it was